### PR TITLE
Allow setting source_location in google_compute_image

### PR DESCRIPTION
**Pull request by Yannic**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/11148

```release-note:enhancement
Allow setting source_location in google_compute_image
```

### CODE REVIEW

**Review by c2thorn — CHANGES_REQUESTED** [review]
Thanks for the contribution @Yannic. Can you add a test demonstrating this field so we can confirm it works? The test file is handwritten here: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_compute_image_test.go.erb

The test you had already written in the previous PR should suffice

**Comment by Yannic** [comment]
@c2thorn thanks for the review and sorry for the delay! Ported the tests over from the previous PR, PTAL

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -7701,7 +7701,6 @@ objects:
         description: |
           The regional or multi-regional Cloud Storage bucket location where the machine image is stored.
         item_type: Api::Type::String
-        output: true
```
Comment (c2thorn) [inline mmv1/products/compute/api.yaml:7704]: Hey @Yannic,

It turns out we can't just remove this attribute. For fields that were API value only, there may be situations where the user has not specified the field, but it is returned by the API anyway. We need to give it the following line in the `compute/terraform.yaml`:
```
      storageLocations: !ruby/object:Overrides::Terraform::PropertyOverride
        default_from_api: true
```
Which makes it default to the API value if the user has not explicitly specified it in their configuration.

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (Yannic) [reply mmv1/products/compute/api.yaml:7704]: thanks, done

Comment (Yannic) [reply mmv1/products/compute/api.yaml:7704]: wait, I misread you message. Let me update the other file

**Review by Yannic — COMMENTED** [review]
PTAL

**Review by c2thorn — CHANGES_REQUESTED** [review]
The added test is failing, and it took me a bit to discover why. The changes to `api.yaml` and `terraform.yaml` are applying to the much to close `MachineImage` resource instead of the `Image` resource. Sorry for the similarity and massive size of the two files.

**Pull request closed without merging**